File: Contrib/SecureBlackbox.NET/Samples/VB.NET/MIMEBlackbox/SecureMail/PassRequestForm.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace SecureMail
    ' <summary>
    ' Summary description for PassRequestForm.
    ' </summary>
    Public Class frmPassRequest
        Inherits System.Windows.Forms.Form

        Private lPrompt As System.Windows.Forms.Label

        Public tbPassphrase As System.Windows.Forms.TextBox

        Private btnOK As System.Windows.Forms.Button

        Private btnCancel As System.Windows.Forms.Button

        Private lKeyInfo As System.Windows.Forms.Label

        ' <summary>
        ' Required designer variable.
        ' </summary>
        Private components As System.ComponentModel.Container = Nothing

        Public Sub New()
            MyBase.New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()
            '
            ' TODO: Add any constructor code after InitializeComponent call
            '
        End Sub

        ' <summary>
        ' Clean up any resources being used.
        ' </summary>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If (Not (components) Is Nothing) Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

        ' <summary>
        ' Required method for Designer support - do not modify
        ' the contents of this method with the code editor.
        ' </summary>
        Private Sub InitializeComponent()
            Me.lPrompt = New System.Windows.Forms.Label
            Me.tbPassphrase = New System.Windows.Forms.TextBox
            Me.btnOK = New System.Windows.Forms.Button
            Me.btnCancel = New System.Windows.Forms.Button
            Me.lKeyInfo = New System.Windows.Forms.Label
            Me.SuspendLayout()
            ' 
            ' lPrompt
            ' 
            Me.lPrompt.Location = New System.Drawing.Point(8, 8)
            Me.lPrompt.Name = "lPrompt"
            Me.lPrompt.Size = New System.Drawing.Size(264, 16)
            Me.lPrompt.TabIndex = 0
            Me.lPrompt.Text = "Passphrase is needed for secret key:"
            Me.lPrompt.TextAlign = System.Drawing.ContentAlignment.MiddleLeft
            ' 
            ' tbPassphrase
            ' 
            Me.tbPassphrase.Location = New System.Drawing.Point(8, 48)
            Me.tbPassphrase.Name = "tbPassphrase"
            Me.tbPassphrase.PasswordChar = Microsoft.VisualBasic.ChrW(42)
            Me.tbPassphrase.Size = New System.Drawing.Size(344, 21)
            Me.tbPassphrase.TabIndex = 1
            Me.tbPassphrase.Text = ""
            ' 
            ' btnOK
            ' 
            Me.btnOK.DialogResult = System.Windows.Forms.DialogResult.OK
            Me.btnOK.Location = New System.Drawing.Point(104, 80)
            Me.btnOK.Name = "btnOK"
            Me.btnOK.TabIndex = 2
            Me.btnOK.Text = "OK"
            ' 
            ' btnCancel
            ' 
            Me.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
            Me.btnCancel.Location = New System.Drawing.Point(184, 80)
            Me.btnCancel.Name = "btnCancel"
            Me.btnCancel.TabIndex = 3
            Me.btnCancel.Text = "Cancel"
            ' 
            ' lKeyInfo
            ' 
            Me.lKeyInfo.Location = New System.Drawing.Point(8, 24)
            Me.lKeyInfo.Name = "lKeyInfo"
            Me.lKeyInfo.Size = New System.Drawing.Size(344, 16)
            Me.lKeyInfo.TabIndex = 4
            ' 
            ' frmPassRequest
            ' 
            Me.AcceptButton = Me.btnOK
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 14)
            Me.CancelButton = Me.btnCancel
            Me.ClientSize = New System.Drawing.Size(362, 111)
            Me.Controls.Add(Me.lKeyInfo)
            Me.Controls.Add(Me.btnCancel)
            Me.Controls.Add(Me.btnOK)
            Me.Controls.Add(Me.tbPassphrase)
            Me.Controls.Add(Me.lPrompt)
            Me.Font = New System.Drawing.Font("Tahoma", 8.25!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(204, Byte))
            Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "frmPassRequest"
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
            Me.Text = "Passphrase request"
            Me.ResumeLayout(False)
        End Sub

        Public Sub Init(ByVal key As SBPGPKeys.TElPGPCustomSecretKey)
            Dim userName As String
            If (Not (key) Is Nothing) Then
                If (TypeOf key Is SBPGPKeys.TElPGPSecretKey) Then
                    If (CType(key, SBPGPKeys.TElPGPSecretKey).PublicKey.UserIDCount > 0) Then
                        userName = CType(key, SBPGPKeys.TElPGPSecretKey).PublicKey.UserIDs(0).Name
                    Else
                        userName = "<no name>"
                    End If
                Else
                    userName = "Subkey"
                End If

                lPrompt.Text = "Passphrase is needed for secret key:"
                lKeyInfo.Text = userName + " (ID=0x" + SBPGPUtils.Unit.KeyID2Str(key.KeyID, True) + ")"
            Else
                lPrompt.Text = "Passphrase is needed to decrypt the message"
                lKeyInfo.Text = ""
            End If
        End Sub
    End Class
End Namespace

File: Contrib/SecureBlackbox.NET/Samples/C#/PGPBlackbox/PGPFiles/PassRequestForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace PGPFilesDemo
{
	/// <summary>
	/// Summary description for PassRequestForm.
	/// </summary>
	public class frmPassRequest : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label lPrompt;
		public System.Windows.Forms.TextBox tbPassphrase;
		private System.Windows.Forms.Button btnOK;
		private System.Windows.Forms.Button btnCancel;
		private System.Windows.Forms.Label lKeyInfo;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public frmPassRequest()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.lPrompt = new System.Windows.Forms.Label();
			this.tbPassphrase = new System.Windows.Forms.TextBox();
			this.btnOK = new System.Windows.Forms.Button();
			this.btnCancel = new System.Windows.Forms.Button();
			this.lKeyInfo = new System.Windows.Forms.Label();
			this.SuspendLayout();
			// 
			// lPrompt
			// 
			this.lPrompt.Location = new System.Drawing.Point(8, 8);
			this.lPrompt.Name = "lPrompt";
			this.lPrompt.Size = new System.Drawing.Size(264, 16);
			this.lPrompt.TabIndex = 0;
			this.lPrompt.Text = "Passphrase is needed for secret key:";
			this.lPrompt.TextAlign = System.Drawing.ContentAlignment.MiddleLeft;
			// 
			// tbPassphrase
			// 
			this.tbPassphrase.Location = new System.Drawing.Point(8, 48);
			this.tbPassphrase.Name = "tbPassphrase";
			this.tbPassphrase.PasswordChar = '*';
			this.tbPassphrase.Size = new System.Drawing.Size(344, 21);
			this.tbPassphrase.TabIndex = 1;
			this.tbPassphrase.Text = "";
			// 
			// btnOK
			// 
			this.btnOK.DialogResult = System.Windows.Forms.DialogResult.OK;
			this.btnOK.Location = new System.Drawing.Point(104, 80);
			this.btnOK.Name = "btnOK";
			this.btnOK.TabIndex = 2;
			this.btnOK.Text = "OK";
			// 
			// btnCancel
			// 
			this.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
			this.btnCancel.Location = new System.Drawing.Point(184, 80);
			this.btnCancel.Name = "btnCancel";
			this.btnCancel.TabIndex = 3;
			this.btnCancel.Text = "Cancel";
			// 
			// lKeyInfo
			// 
			this.lKeyInfo.Location = new System.Drawing.Point(8, 24);
			this.lKeyInfo.Name = "lKeyInfo";
			this.lKeyInfo.Size = new System.Drawing.Size(344, 16);
			this.lKeyInfo.TabIndex = 4;
			// 
			// frmPassRequest
			// 
			this.AcceptButton = this.btnOK;
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 14);
			this.CancelButton = this.btnCancel;
			this.ClientSize = new System.Drawing.Size(362, 111);
			this.Controls.Add(this.lKeyInfo);
			this.Controls.Add(this.btnCancel);
			this.Controls.Add(this.btnOK);
			this.Controls.Add(this.tbPassphrase);
			this.Controls.Add(this.lPrompt);
			this.Font = new System.Drawing.Font("Tahoma", 8.25F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(204)));
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
			this.MaximizeBox = false;
			this.MinimizeBox = false;
			this.Name = "frmPassRequest";
			this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
			this.Text = "Passphrase request";
			this.ResumeLayout(false);

		}
		#endregion

		public void Init(SBPGPKeys.TElPGPCustomSecretKey key)
		{
			string userName;
			if (key != null) 
			{
				if (key is SBPGPKeys.TElPGPSecretKey) 
				{
					if (((SBPGPKeys.TElPGPSecretKey)key).PublicKey.UserIDCount > 0) 
					{
						userName = ((SBPGPKeys.TElPGPSecretKey)key).PublicKey.get_UserIDs(0).Name;
					} 
					else 
					{
						userName = "<no name>";
					}
				} 
				else 
				{
					userName = "Subkey";
				}
				lPrompt.Text = "Passphrase is needed for secret key:";
				lKeyInfo.Text = userName + " (ID=0x" + SBPGPUtils.Unit.KeyID2Str(key.KeyID(), true) + ")";
			} 
			else 
			{
				lPrompt.Text = "Passphrase is needed to decrypt the message";
				lKeyInfo.Text = "";
			}
		}
	}
}
